golang-sample/cloudbuild.yaml

```diff
@@ -0,0 +1,61 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+steps:
+  # Initialize the module
+  - name: golang
+    args: ['go', 'mod', 'init', 'github.com/GoogleCloudPlatform/cloud-build-samples/golang']
+
+  # Clean up dependencies
+  - name: golang
+    args: ['go', 'mod', 'tidy']
+
+  # Build application
+  - name: golang
+    args: ['go', 'build', '.']
+
+  # Run tests and save to file
+  - name: golang
+    entrypoint: /bin/bash
+    args: 
+      - -c
+      - |
+        go test . -v -json > ${SHORT_SHA}_test_log.json
```

Comment:
Leaning on shell utilities is aligned with Go practice. Not sure if you want to keep with the JSON output -- most test processing tools expect xunit style.

Here's how the tests are executed and captured in golang-samples:

```
2>&1 go test -timeout $TIMEOUT -v "${1:-./...}" | tee sponge_log.log
/go/bin/go-junit-report -set-exit-code < sponge_log.log > raw_log.xml
```

* The `tee` utility means the test output is displayed in the build log as well as written to the file for further processing.
* The error output redirection ensures any error messages are written to the log file.